DigitalLearningSolutions.Web/Helpers/RegistrationEmailValidator.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace DigitalLearningSolutions.Web.Helpers
+{
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+    public static class RegistrationEmailValidator
+    {
+        private const string WrongEmailForCentreErrorMessage =
+            "This email address does not match the one held by the centre";
+
+        private const string EmailInUseErrorMessage =
+            "A user with this email address is already registered; if this is you, " +
+            "please log in at this centre via the My Account page";
+
+        public static void ValidateEmailAddresses(
+            PersonalInformationViewModel model,
+            ModelStateDictionary modelState,
+            IUserService userService,
+            ICentresService? centresService = null
+        )
+        {
+            if (!model.Centre.HasValue)
+            {
+                return;
+            }
+
+            var isRegisterAdminJourney = centresService != null;
+            var primaryEmailIsValid = !modelState.HasError(nameof(PersonalInformationViewModel.PrimaryEmail)) &&
+                                      model.PrimaryEmail != null;

```

Comment:
Since this method is called in the GET as well as the POST (since the GET might be loaded with info populated from temp data), I had to implement this check to avoid hitting an error when calling a service method below when called from the GET when the emails are null. I change the variable names so this is clearer.